routes/admin/users.rb

```diff
@@ -446,6 +455,45 @@
         flash(:alert, 'Passwords do not match', 'Sorry, your passwords do not match.')
         redirect back
     end
+    if params[:university_status] != 'Non-NU Student (All Other Institutions)'
+        # Try to get user nuid from directory api
+        begin
+            content = fetch_final_content("https://directory.unl.edu/people/#{params[:username]}?format=json")
```

Comment:
When a user logs in we will want to check if their NUID is empty, and if it is then we should try to get it from directory. So we can get that data for users that have already logged in.

If the user isn't a real person or something, we don't want to display an error if we can't get their NUID. We also might want to keep track of the number of failed attempts and if that is above a threshold we can stop trying.